src/data_base.cpp

```diff
@@ -231,7 +231,7 @@ code data_base::catalog(const transaction& tx)
 
     // Existence check prevents duplicated indexing.
     if (!settings_.index_addresses || tx.metadata.existed)
-        return ec;
+        return error::success;
```

Comment:
This causes a reconstruction of the returned `code` object, whereas the original allowed the compiler to move the value.